bin/blackmagic.py

```diff
@@ -592,44 +599,67 @@ def search(self, request, query):
 
         request.ret(packages_list)
 
+    @only_if_initialized
+    @remote
+    def desktop_environment(self, request, xfce4):
+        self.image['xfce4'] = bool(xfce4)
+        self.db.images.replace_one({'_id': self.image['id']}, self.image, True)
+        request.ret(READY)
+
     @only_if_initialized
     @remote
     def resolve(self, request, packages_list):
         self.image['selected_packages'] = packages_list
         self.db.images.replace_one({'_id': self.image['id']}, self.image, True)
         resolver_env = self.image['resolver_env']
 
-        command_line = [
-            'apt-get', 'install', '--no-act', '-qq',
-            '-o', 'APT::Architecture=all',
-            '-o', 'APT::Architecture=' + self._arch,
-            '-o', 'APT::Default-Release=' + self._suite,
-            '-o', 'Dir=' + resolver_env,
-            '-o', 'Dir::State::status=' + resolver_env +
-                  '/var/lib/dpkg/status'
-        ] + packages_list
-
-        data = yield util.execute_async(command_line)
-
-        # The output of the above command line will look like the
-        # following set of lines:
-        # NOTE: This is only a simulation!
-        #       apt-get needs root privileges for real execution.
-        #       Keep also in mind that locking is deactivated,
-        #       so don't depend on the relevance to the real current situation!
-        # Inst libgdbm3 (1.8.3-13.1 Debian:8.4/stable [armhf])
-        # Inst libssl1.0.0 (1.0.1k-3+deb8u4 Debian:8.4/stable [armhf])
-        # Inst libxml2 (2.9.1+dfsg1-5+deb8u1 Debian:8.4/stable [armhf])
-        # ...
-        # Conf libgdbm3 (1.8.3-13.1 Debian:8.4/stable [armhf])
-        # Conf libssl1.0.0 (1.0.1k-3+deb8u4 Debian:8.4/stable [armhf])
-        # Conf libxml2 (2.9.1+dfsg1-5+deb8u1 Debian:8.4/stable [armhf])
-        # ...
-        packages_to_be_installed = self.inst_pattern.findall(str(data))
-        dependencies = set(packages_to_be_installed) - set(packages_list)
-
-        # Python sets are not JSON serializable
-        request.ret(list(dependencies))
+        packages_list_sort = sorted(packages_list)
+        packages_list_str = '-'.join(packages_list_sort)
+        os_name = self._os
+        key_cashe = os_name + '-' + packages_list_str
+
+        key_cashe_value = self.redis_conn.get(key_cashe)
+
+        if key_cashe_value:
+            dependencies_cache = json.loads(key_cashe_value)
+            LOGGER.info(dependencies_cache)
```

Comment:
@leocoan please remove `LOGGER.info(dependencies_cache)`. I suppose you only need it for local debugging